matlab/+caffe/private/caffe_.cpp

```diff
@@ -45,7 +45,11 @@ void mxCHECK_FILE_EXIST(const char* file) {
 static vector<shared_ptr<Solver<float> > > solvers_;
 static vector<shared_ptr<Net<float> > > nets_;
 // init_key is generated at the beginning and everytime you call reset
-static double init_key = static_cast<double>(caffe_rng_rand());
```

Comment:
@mnidza Sorry about the typo...
It seems that MATLAB halt on line 112 in `common.cpp` when `cublasCreate(&cublas_handle_)` is called. Set `CPU_ONLY` will get rid of this part, so CPU-only build have no issues.